elide-core/src/main/java/com/yahoo/elide/Elide.java

```diff
@@ -276,19 +276,24 @@ protected ElideResponse buildErrorResponse(HttpStatusException error, boolean is
             ErrorObjects errors = ErrorObjects.builder().addError()
                     .withDetail(isVerbose ? error.getVerboseMessage() : error.toString()).build();
             JsonNode responseBody = mapper.getObjectMapper().convertValue(errors, JsonNode.class);
-            return buildResponse(Pair.of(error.getStatus(), responseBody));
+            return buildResponse(Pair.of(error.getStatus(), responseBody), error);
         }
-        return buildResponse(isVerbose ? error.getVerboseErrorResponse() : error.getErrorResponse());
+        return buildResponse(isVerbose ? error.getVerboseErrorResponse() : error.getErrorResponse(), error);
     }
 
+    @Deprecated
     protected ElideResponse buildResponse(Pair<Integer, JsonNode> response) {
+        return buildResponse(response, null);
+    }
+
+    protected ElideResponse buildResponse(Pair<Integer, JsonNode> response, Throwable failureReason) {
         try {
             JsonNode responseNode = response.getRight();
             Integer responseCode = response.getLeft();
             String body = responseNode == null ? null : mapper.writeJsonApiDocument(responseNode);
-            return new ElideResponse(responseCode, body);
+            return new ElideResponse(responseCode, body, failureReason);
         } catch (JsonProcessingException e) {
-            return new ElideResponse(HttpStatus.SC_INTERNAL_SERVER_ERROR, e.toString());
+            return new ElideResponse(HttpStatus.SC_INTERNAL_SERVER_ERROR, e.toString(), e);
```

Comment:
What about graphQL?

Reply:
There is no real analog to `ElideResponse` in GraphQL because we're leveraging GraphQLJava. The `Elide` class drives JsonAPI and is ignored by GraphQL, also the `GraphQLEndpiont` can't be customized in the way that `JsonApiEndpoint` can.